easy-header-footer-adapter/src/main/java/com/rubengees/easyheaderfooteradapter/EasyHeaderFooterAdapter.java

```diff
@@ -63,8 +63,10 @@ public void onItemRangeRemoved(int positionStart, int itemCount) {
             }
 
             public void onItemRangeMoved(int fromPosition, int toPosition, int itemCount) {
-                notifyItemRangeChanged(getDelegatedPosition(fromPosition),
-                        getDelegatedPosition(toPosition) + itemCount);
+                // fix bug. Because fromPosition may bigger then toPosition.
+                // For now, itemCount must be 1. So just call notifyItemMoved
+                // 目前 recycleView源码中，itemCount只能是1，因为只能通过adapter.notifyItemMoved触发。
+                notifyItemMoved(getDelegatedPosition(fromPosition), getDelegatedPosition(toPosition));
```

Comment:
This only applies to a single item. What if more items are in between `fromPosition` and `toPosition`?

Reply:
Quite agree with you -- "This only applies to a single item. "
But in some cases, fromPosition == 2, toPosition == 1 ,  itemCount == 1.
I don't think your code can handle this case.

Reply:
"What if more items are in between fromPosition and toPosition?"
For now, this will not happened after I read the source code of RecyclerView.java

when the mothed of notifyItemMoved be called:

public void notifyItemMoved(int fromPosition, int toPosition) {
            for (int i = mObservers.size() - 1; i >= 0; i--) {
                mObservers.get(i).onItemRangeMoved(fromPosition, toPosition, 1);
            }
        }

Reply:
After deep working, I finally use notifyDateSet() in all method except onItemChange. 
Looking forward your replay.

Reply:
You should not use `notifyDataSetChanged` if not absolutely necessary. It requires stable ids to work and checks the entire dataset before updating. Using exact `notify` methods should always be preferred, since those cause the `RecyclerView` to only update the changed once (More on this [here](https://stackoverflow.com/questions/33789345/whats-better-notifydatasetchanged-or-notifyitemchanged-in-loop)).

I would be fine with merging this, if the old behaviour is preserved for the case `itemCount > 0`. Even if this case can not happen now, it may in the future.

Reply:
```java
if (itemCount == 1) {
    notifyItemMoved(getDelegatedPosition(fromPosition), getDelegatedPosition(toPosition));
} else if (itemCount > 1) {
    notifyItemRangeChanged(getDelegatedPosition(fromPosition),
            getDelegatedPosition(toPosition) + itemCount);
}